test/junit/scala/reflect/ClassTagTest.scala

```diff
@@ -27,6 +27,13 @@ class ClassTagTest {
   @Test def checkBoolean = assertTrue(checkNotInt[Boolean](false))
   @Test def checkUnit    = assertTrue(checkNotInt[Unit]   ({}))
 
+  @Test def unapplyCheckDouble = {
+    val double = 1.0
+    val ct = classTag[Double]
+    val clazz = ct.unapply("123")
+    println(clazz)
```

Comment:
Why `println` ? Some kind of assertion would be better.

Reply:
sorry, this test case is just used to help me test whether the unapply can delegate to correct child class. Will remove it in final PR